### PR TITLE
feat(chrome): replace background color with `hue` prop

### PR DESCRIPTION
**Pull request by jzempel**
- [x] **BREAKING CHANGE?** 

## Description

The `hue` prop is used to alter `Nav` and `SubNav` color treatments.

## Detail

Additionally, this PR includes opacity fixes for navigation hover/focus states.

Demo is pre-published for review: https://vigorous-davinci-819228.netlify.com/chrome/

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [ ] :nail_care: ~view component styling is based on a Garden CSS
      component~
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [ ] :wheelchair: ~analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver~
- [x] :guardsman: includes new unit tests
- [ ] :memo: ~tested in Chrome, Firefox, Safari, Edge, and IE11~

### CODE REVIEW

File: packages/chrome/src/elements/Chrome.tsx [hunk 1]
```diff
@@ -5,12 +5,55 @@
...
-import React, { HTMLAttributes } from 'react';
+import React, { HTMLAttributes, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { ThemeProps, DefaultTheme } from 'styled-components';
+import readableColor from 'polished/lib/color/readableColor';
+import { withTheme, DEFAULT_THEME, getColor } from '@zendeskgarden/react-theming';
+import { ChromeContext } from '../utils/useChromeContext';
 import { StyledChrome } from '../styled';
 
+interface IChromeProps extends HTMLAttributes<HTMLDivElement> {
```
Comment (austin94) [inline packages/chrome/src/elements/Chrome.tsx]: This context only affects the `Nav` item, right? Could we move this prop and the context provider to the `Nav` component to limit its scope?

Comment (jzempel) [reply packages/chrome/src/elements/Chrome.tsx]: No, it affects `SubNav` (and any future Chrome-related components that should be affected by the top-level hue).

**Comment by jzempel** [comment]
@allisonacs @austin94 sorry y'all. The last demo deploy failed. It's deployed to a new location now: https://vigorous-davinci-819228.netlify.com/chrome/. You'll see the subnav color update along with a "Hue" modification.

**Comment by jzempel** [comment]
Thanks, @allisonacs that's certainly gratifying :wink:

I'm not super satisfied with the `readableColor` utility that we're using from Polished JS to determine whether to show light-on-dark/dark-on-light. They [claim](https://github.com/styled-components/polished/blob/0ac0bbcf159f975907841aedb10bbe1944250c5f/src/color/readableColor.js#L6) to be using the luminance algorithm described by https://www.w3.org/TR/WCAG20-TECHS/G18.html.

I've had better results from TinyColor whose `isLight/isDark` [implementation](https://github.com/bgrins/TinyColor/blob/96592a5cacdbf4d4d16cd7d39d4d6dd28da9bd5f/tinycolor.js#L52-L60) is based on a W3-recommended brightness algorithm: https://www.w3.org/TR/AERT/#color-contrast.

As you say, we can dial-in the preferred approach later on.

**Pull request merged**